opsmngr/feature_control_policies.go

```diff
@@ -0,0 +1,138 @@
+
+// Agent represents an Ops Manager agent
+type FeaturePolicy struct {
+	ID                       string                    `json:"_id"`
+	Created                  string                    `json:"created"`
+	Updated                  string                    `json:"updated,omitempty"`
+	ExternalManagementSystem *ExternalManagementSystem `json:"externalManagementSystem,omitempty"`
+	Policies                 []*Policy                 `json:"policies,omitempty"`
+}
+
+// ExternalManagementSystem contains parameters for the external system that manages this Ops Manager Project
+type ExternalManagementSystem struct {
+	Name     string `json:"name"`
+	SystemID string `json:"systemId"`
+	Version  string `json:"version"`
+}
+
+// Policy contains policies that the external system applies to this Ops Manager Project
+type Policy struct {
+	Policy         string   `json:"policy,omitempty"`
+	DisabledParams []string `json:"disabledParams,omitempty"`
+}
+
+// List retrieves the Feature Control Policies for one Project.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/controlled-features/get-controlled-features-for-one-project/
+func (s *FeatureControlPoliciesServiceOp) List(ctx context.Context, groupID string, opts *atlas.ListOptions) (*FeaturePolicy, *atlas.Response, error) {
+	if groupID == "" {
+		return nil, nil, atlas.NewArgError("groupID", "must be set")
+	}
+	path := fmt.Sprintf(controlledFeaturesBasePath, groupID)
+
+	path, err := setQueryParams(path, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.Client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(FeaturePolicy)
+	resp, err := s.Client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+// Update updates the Feature Control Policies for one Project.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/controlled-features/update-controlled-features-for-one-project/
+func (s *FeatureControlPoliciesServiceOp) Update(ctx context.Context, groupID string, policy *FeaturePolicy) (*FeaturePolicy, *atlas.Response, error) {
+	if groupID == "" {
+		return nil, nil, atlas.NewArgError("groupID", "must be set")
+	}
+	path := fmt.Sprintf(controlledFeaturesBasePath, groupID)
+
+	req, err := s.Client.NewRequest(ctx, http.MethodPut, path, policy)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(FeaturePolicy)
+	resp, err := s.Client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return root, resp, err
+}
+
+// ListAll retrieves the available Feature Control Policies.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/controlled-features/get-all-feature-control-policies/
+func (s *FeatureControlPoliciesServiceOp) ListAll(ctx context.Context, opts *atlas.ListOptions) (*FeaturePolicy, *atlas.Response, error) {
```

Comment:
The naming is confusing here. `ListAll` is retrieves all supported policies by Ops Manager. `List` is retrieving the policies that have seen set on a project.

Instead of `ListAll`, I'd use a convention like `Supported` or `Available` For reference, the endpoint is `GET /groups/availablePolicies`